### ISSUE TITLE
Pretty stack trace for standalone break package and debugger

### ISSUE THREAD

**Issue by kenbowen** [AppliedLogicSystems/ALSProlog#42]
Create a “pretty” stack trace mechanism which focuses on display of the user code, and which would be made available in both the standalone break package and the debugger as option “u - pretty stack trace”.  The code should be as close to common as possible between the standalone break package and the debugger; it should emulate (as near as possible) the steps taken by the debugger to avoid showing (noshow) both debugger internals and the internals of builtins.  Cf. the closed #37 for concerns.

**Comment by chuckhoupt** [comment]
Another idea is to "enhance" the existing stack trace. For example, user frames could be displayed in bold, while builtin frames are de-emphasized with light grey. Also, a GUI stack display could collapse runs of builtin frames into a single line (to again de-emphasize them). Clicking on the collapsed line would expand it, for those who want to see every detail.

**Comment by kenbowen** [comment]
Good idea.  Also likely that it could be configurable with a dynamic flag which could affect both the tty and guy versions; e.g., Never Show Details, ......, Always Show Details.
